TOPIC 1/discount/DiscountHandler.java

```diff
@@ -0,0 +1,13 @@
+package discount;
+
+import shopping.Offer;
+
+public abstract class DiscountHandler {
+    DiscountHandler successor;
```

Comment:
Veo que aplicaste un Chain of responsibility. No esta en la consigna, pero bien pensado e implementado igualmente!